File: 01 arrays/exercise - containsDuplicate.py
# 217. Contains Duplicate
# https://leetcode.com/problems/contains-duplicate/description/

def containsDuplicate(nums):
  if (not nums or 0 == len(nums)):
    return False

  for i in range(len(nums)):
    val = nums[i]
    for j in range(i + 1, len(nums)):
      if nums[j] == val:
        return True

  return False


def containsDuplicate2(nums):
  if (not nums or 0 == len(nums)):
    return False

  lst = set()

  for i in range(len(nums)):
    val = nums[i]
    if val in lst:
      return True
    lst.add(val)

  return False

File: 01 arrays/exercise - containsDuplicate.js
// 217. Contains Duplicate
// https://leetcode.com/problems/contains-duplicate/description/

var containsDuplicate = function(nums) {
  if (!nums || nums.length == 0)
    return false;

  var hash = {};

  for (let i = 0; i < nums.length; i++) {
    let val = nums[i];

    if (hash[val])
      return true;

    hash[val] = true;
  }

  return false;
}

var containsDuplicate2 = function(nums) {
  if (!nums || nums.length == 0)
    return false;

  for (let i = 0; i < nums.length; i++) {
    val = nums[i];
    for (let j = i + 1;j < nums.length; j++) {
      if (val == nums[j])
        return true;
    }
  }

  return false;
}
